### PR TITLE
Only test view function, not decorated function in lti_launches, content_item_selection view tests

### PR DESCRIPTION
**Pull request by robertknight**
The `lti_launches` and `content_item_selection` view functions are
wrapped by a nested series of decorators which handle authorization and,
when provisioning is enabled, creation of users and groups in h that
correspond to the lms course.

These decorators are hard to mock/stub in unit tests which are only
concerned with the functionality of the view callable itself. This
problem is about to be exacerbated as provisioning becomes the default
experience and all the provisioning-related decorators therefore get
activated instead of skipping their logic.

The preferred way to resolve this problem would be to refactor the view
callable to perform provisioning setup and authorization in a way that
is more easily mocked, eg. using calls to services.

As an intermediate step, this PR refactors the tests for several view
functions to unwrap the decorated function and just test the view callables.
This involves:

 - Modifying the existing decorators to use `functools.wraps`, which
   adds a `__wrapped__` attr to the decorated function which is a
   reference to the original function.

 - Unwrapping the decorated function in the tests using a helper
   `unwrap` function.

 - Modifying the tests to pass any additional arguments which the
   wrapped function requires.

In the case of the existing `content_item_selection` tests, this
revealed that the existing test had nothing to do with the view function
itself but was in fact a test for logic in `lms.util.authorize_lms`.

### CODE REVIEW

File: tests/lms/views/test_content_item_selection.py [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
 
+# The `content_item_selection` view function is wrapped in a series of
+# decorators which handle authorization and creating the user/group for the
+# current course.
+#
+# In these tests we only want to test the view function itself, so extract that
+# from the decorated function.
+#
+# nb. This stack of decorators makes it hard to test. It would be much better to
+# use service method calls to handle the work done by decorators.
```
Comment (seanh) [inline tests/lms/views/test_content_item_selection.py]: Can we remove the last paragraph of this comment? I'm not sure that view body code calling services is actually how we'll refactor this

File: tests/lms/views/test_lti_launches.py [hunk 2]
```diff
@@ -1,28 +1,43 @@
...
+
+
+# The `lti_launches` view function is wrapped in a series of decorators which
+# handle authorization and creating the user/group for the current course.
+#
+# In these tests we only want to test the view function itself, so extract that
+# from the decorated function.
+#
+# nb. This stack of decorators makes it hard to test. It would be much better to
+# use service method calls to handle the work done by decorators.
```
Comment (seanh) [inline tests/lms/views/test_lti_launches.py]: Can we remove the last paragraph of this comment? I'm not sure that view body code calling services is actually how we'll refactor this

File: tests/lms/views/test_content_item_selection.py (hunk 1 above)
Comment (robertknight) [reply tests/lms/views/test_content_item_selection.py]: Done

File: tests/lms/views/test_lti_launches.py (hunk 2 above)
Comment (robertknight) [reply tests/lms/views/test_lti_launches.py]: Done

**Pull request merged**